packages/client-core/src/admin2/components/user/AddEditUserModal.tsx

```diff
@@ -0,0 +1,200 @@
+Original Code is the Ethereal Engine team.
+All portions of the code written by the Ethereal Engine team are Copyright © 2021-2023 
+Ethereal Engine. All Rights Reserved.
+*/
+
+import { PopoverState } from '@etherealengine/client-core/src/common/services/PopoverState'
+import {
+  AvatarID,
+  ScopeType,
+  UserData,
+  UserName,
+  UserType,
+  avatarPath,
+  scopeTypePath,
+  userPath
+} from '@etherealengine/common/src/schema.type.module'
+import { useHookstate } from '@etherealengine/hyperflux'
+import { useFind, useMutation } from '@etherealengine/spatial/src/common/functions/FeathersHooks'
+import Button from '@etherealengine/ui/src/primitives/tailwind/Button'
+import Input from '@etherealengine/ui/src/primitives/tailwind/Input'
+import Label from '@etherealengine/ui/src/primitives/tailwind/Label'
+import LoadingCircle from '@etherealengine/ui/src/primitives/tailwind/LoadingCircle'
+import Modal from '@etherealengine/ui/src/primitives/tailwind/Modal'
+import MultiSelect from '@etherealengine/ui/src/primitives/tailwind/MultiSelect'
+import Select, { SelectProps } from '@etherealengine/ui/src/primitives/tailwind/Select'
+import React, { useEffect } from 'react'
+import { useTranslation } from 'react-i18next'
+import AccountIdentifiers from './AccountIdentifiers'
+
+const getDefaultErrors = () => ({
+  name: '',
+  avatarId: ''
+})
+
+export default function AddEditUserModal({ user }: { user?: UserType }) {
+  const { t } = useTranslation()
+
+  const userMutation = useMutation(userPath)
+  const avatarsQuery = useFind(avatarPath, {
+    query: {
+      action: 'admin'
+    }
+  })
+  const avatarOptions =
+    avatarsQuery.status === 'success'
+      ? [
+          { name: t('admin:components.user.selectAvatar'), value: '', disabled: true },
+          ...avatarsQuery.data.map((av) => ({ name: av.name, value: av.id }))
+        ]
+      : [{ name: t('common:select.fetching'), value: '', disabled: true }]
+
+  const scopeTypesQuery = useFind(scopeTypePath, {
+    query: {
+      paginate: false
+    }
+  })
+  const scopeTypeOptions: SelectProps['options'] =
+    scopeTypesQuery.status === 'success'
+      ? [
+          { name: t('admin:components.user.selectScopes'), value: '', disabled: true },
+          ...scopeTypesQuery.data.map((st) => ({ name: st.type, value: st.type }))
+        ]
+      : [{ name: t('common:select.fetching'), value: '', disabled: true }]
+
+  if (user) {
+    for (const scope of user.scopes || []) {
+      const scopeExists = scopeTypeOptions.find((st) => st.value === scope.type)
+      if (!scopeExists) {
+        scopeTypeOptions.push({ name: scope.type, value: scope.type })
+      }
+    }
+
+    if (!avatarOptions.find((av) => av.value === user.avatarId)) {
+      avatarOptions.push({ name: user.avatar.name || user.avatarId, value: user.avatarId })
+    }
+  }
+
+  const submitLoading = useHookstate(false)
+  const errors = useHookstate(getDefaultErrors())
```

Comment:
we can probably use the value directly instead of a function. are we using this function elsewhere?

Reply:
Had added it for resetting/clearing the errors when form is resubmitted. Hookstate takes the reference of the object when resetting it.

this does not work:
```ts
errors.set(defaultErrors);
```
but this does:
```ts
errors.set(getDefaultErrors());
```
because the `getDefaultErrors()` function creates a new object each time.